sympy/sets/fancysets.py

```diff
@@ -544,27 +563,38 @@ def __new__(cls, *args):
     Either the start or end value of the Range must be finite.'''))
 
         if start.is_infinite:
-            if step*(stop - start) < 0:
-                start = stop = S.One
+            # null Range
+            if (step*(stop - start) < 0) == True:
+                start = end = S.Zero
+                step = S.One
             else:
-                end = stop
-        if not start.is_infinite:
-            ref = start if start.is_finite else stop
+                raise ValueError("Non empty Range with infinite start is not allowed."
```

Comment:
I would recommend, "Range must start with a finite value or symbol."

(For display puposes, one could use `Symbol("oo")` for the start...)

Reply:
Well, after disallowing `Range` with infinite start, many other modules are giving failing tests. Would it be still good to remove it? What do you say?
Something like, `Range(oo, 2, -2)` is ambiguous, so reversing it in the constructor and then returning a `Range` with modified parameters will also not be a good idea. Any comments? Any workaround? Or the only option is to modify the failing tests for the introduced change.

Reply:
Or don't break with master and continue to allow only `Range(-oo, n)` with step of 1.

Reply:
But the tests that are failing are from master where a non-unit step is allowed as in `Range(oo, -2, -2)`. 

I would first focus on the stats failure. What result is that giving now? The other failures are just printing tests.

Reply:
The `Range(-oo, 0, 1)` comes into picture during the computations which caused the failure in `sympy.stats`. So, if we allow, `Range(-oo, n)` then there would be no problem with that test.

Reply:
I mean, "What is the result that the test is generating now?" It's not matching the current expectation expression; it's not raising an error for starting with `-oo`, is it?

Reply:
`Probability(z < 0)` that is un-evaluated result. Probably it's not throwing an error because of try-except block in `sympy.stats.drv.DiscretePSpace.probability`. However, that try-except catches `NotImplementedError`. I don't know why is it catching `ValueError` but still I suspect the above as the reason.

Reply:
Ok. So can you find out which routine is generating the Range with the infinite start and see how easy it would be to reverse that range?

Reply:
`sympy.stats.drv.DiscretePSpace.where.set` generates `Range(-oo, 0, 1)` as answer. I will take a look at it.